bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/util/ThingHelper.java

```diff
@@ -81,7 +81,7 @@ public static boolean equals(Thing a, Thing b) {
     private static String toString(List<Channel> channels) {
         List<String> strings = new ArrayList<>(channels.size());
         for (Channel channel : channels) {
-            strings.add(channel.getUID().toString() + '#' + channel.getAcceptedItemType());
+            strings.add(channel.getUID().toString() + '#' + channel.getAcceptedItemType() + '#' + channel.getTriggerType());
```

Comment:
Yes, but this whole string creation is done only for comparing the channels of the things. What's the problem with the null in there?